src/wp-includes/media.php

```diff
@@ -5564,6 +5686,123 @@ function wp_get_loading_attr_default( $context ) {
 	return 'lazy';
 }
 
+/**
+ * Gets loading optimization attributes.
+ *
+ * @since 6.3.0
+ *
+ * @param string  $tag_name The tag name.
+ * @param array[] $attr     Array of the attributes for the tag.
+ * @param string  $context  Context for the element for which the loading optimization attribute is requested.
+ *
+ * @return array[] Loading optimization attributes.
+ */
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
+	global $wp_query;
+
+	$loading_attrs = array();
+
+	// For now this function only supports images.
+	if ( 'img' !== $tag_name ) {
+		return $loading_attrs;
+	}
+
+	// For any resources, width and height must be provided, to avoid layout shifts.
+	if ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) {
+		return $loading_attrs;
+	}
+
+	if ( isset( $attr['loading'] ) && 'lazy' === $attr['loading'] ) {
+		if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority']
+		) {
+			_doing_it_wrong(
+				__FUNCTION__,
+				sprintf(
+					/* translators: %s: fetchpriority="high". */
+					__( 'An image cannot be lazy-loaded and assigned %s at the same time.' ),
+					'<code>fetchpriority="high"</code>'
+				),
```

Comment:
Minor wording suggestion similar to above:
```suggestion
				__( 'An image should not be lazy-loaded and marked as high priority at the same time.' ),
```